### PR TITLE
new Developer overlay for choosing provider

### LINKED ISSUE #804: replace /provider with a developer overlay dropdown to choose metamask on localhost
**Is your feature request related to a problem? Please describe.**
It is often inconvenient to go to /provider, and then click back, especially after a reload is necessary

**Describe the solution you'd like**
A simple overlay dropdown that allows choosing Metamask, and shows which network is selected, which would work on any page, whether it is dashboard or demo or another future page

**Describe alternatives you've considered**
Haven't thought of any alternatives

### PR DESCRIPTION
**Pull request by cellog**
# Description

un-conflicted version of #816 fixes #804 

See #816 for screenshot and checklist. (whew)

### CODE REVIEW

**Comment by cellog** [comment]
superseded, closing

**Comment by cellog** [comment]
(oops closed wrong one)

**Comment by cellog** [comment]
Just need to add the new snapshots and this is ready for merge. Feel free to review the code before then

File: unlock-app/src/propTypes.js [hunk 1]
```diff
@@ -30,6 +30,8 @@ export const conversion = PropTypes.objectOf(PropTypes.number)
 
 export const children = PropTypes.shape({})
 
+export const callback = PropTypes.func
```
Comment (julien51) [inline unlock-app/src/propTypes.js]: Is that necessary? I mean, a `callback` is nothing specific to Unlock?

File: unlock-app/src/components/developer/DeveloperOverlay.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        </Inner>
+      </Overlay>
+    </NoSSR>
+  )
+}
+
+DeveloperOverlay.propTypes = {
+  config: UnlockPropTypes.configuration.isRequired,
+  selected: UnlockPropTypes.provider,
+  setProvider: UnlockPropTypes.callback.isRequired,
```
Comment (julien51) [inline unlock-app/src/components/developer/DeveloperOverlay.js]: I'd argue that this is actually not even a `callback` in the JS sense, but really a regular function invoked on user actions ;) Let's just stick to `PropTypes.func`!

File: unlock-app/src/components/developer/DeveloperOverlay.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+export const mapStateToProps = state => {
+  return { selected: state.provider }
+}
+
+export default withConfig(
+  connect(
+    mapStateToProps,
+    { setProvider }
```
Comment (julien51) [inline unlock-app/src/components/developer/DeveloperOverlay.js]: Nice syntax ;)

File: unlock-app/src/propTypes.js (hunk 1 above)
Comment (cellog) [reply unlock-app/src/propTypes.js]: I wasn't sure, since there are some other logical aliases in the file :) I'm happy using .func

**Comment by cellog** [comment]
@julien51 all set for final review

**Comment by cellog** [comment]
screenshot

**Pull request merged**